src/osm_callbacks.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+#ifndef TIMESCALEDB_OSM_CALLBACKS_H
+#define TIMESCALEDB_OSM_CALLBACKS_H
+
+#include <postgres.h>
+#include <catalog/objectaddress.h>
+
+typedef int (*chunk_insert_check_hook_type)(Oid ht_oid, int64 range_start, int64 range_end);
+typedef void (*hypertable_drop_hook_type)(const char *schema_name, const char *table_name);
```

Comment:
if this is only about `deletion` as the PR title suggest; then what does `insert` doing here?

I feel that only declared `drop` + `insert` ; but I think other things may happen as well...what about: create, delete/etc ?

other parts of the PR makes me wonder if this is a general `hypertable` lifecycle hook...

Reply:
Insert callback is for checking for intersections with chunks managed by OSM. This check already was in the code, I'm not adding it, see code changes in `chunks.c`. We just decided it would be nicer to have one struct with all the callbacks OSM might need.